packages/features/bookings/Booker/Booker.tsx

```diff
@@ -0,0 +1,213 @@
+import type { MotionStyle } from "framer-motion";
+import { LazyMotion, domAnimation, m, AnimatePresence } from "framer-motion";
+import { Fragment, useEffect, useRef } from "react";
+import StickyBox from "react-sticky-box";
+import { shallow } from "zustand/shallow";
+
+import classNames from "@calcom/lib/classNames";
+import useGetBrandingColours from "@calcom/lib/getBrandColours";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import useMediaQuery from "@calcom/lib/hooks/useMediaQuery";
+import { Logo, ToggleGroup, useCalcomTheme } from "@calcom/ui";
+import { Calendar, Columns, Grid } from "@calcom/ui/components/icon";
+
+import { AvailableTimeSlots } from "./components/AvailableTimeSlots";
+import { Away } from "./components/Away";
+import { BookEventForm } from "./components/BookEventForm";
+import { DatePicker } from "./components/DatePicker";
+import { EventMeta } from "./components/EventMeta";
+import { LargeCalendar } from "./components/LargeCalendar";
+import { BookerSection } from "./components/Section";
+import { fadeInUp, fadeInLeft, resizeAnimationConfig } from "./config";
+import { useBookerStore, useInitializeBookerStore } from "./store";
+import type { BookerLayout, BookerProps } from "./types";
+import { useEvent } from "./utils/event";
+
+const useBrandColors = ({ brandColor, darkBrandColor }: { brandColor?: string; darkBrandColor?: string }) => {
+  const brandTheme = useGetBrandingColours({
+    lightVal: brandColor,
+    darkVal: darkBrandColor,
+  });
+  useCalcomTheme(brandTheme);
+};
+
+const BookerComponent = ({ username, eventSlug, month, rescheduleBooking }: BookerProps) => {
+  const { t } = useLocale();
+  const isMobile = useMediaQuery("(max-width: 768px)");
+  const isTablet = useMediaQuery("(max-width: 1024px)");
+  const timeslotsRef = useRef<HTMLDivElement>(null);
+  const StickyOnDesktop = isMobile ? "div" : StickyBox;
+  const rescheduleUid =
+    typeof window !== "undefined" ? new URLSearchParams(window.location.search).get("rescheduleUid") : null;
+  const event = useEvent();
+  const [layout, setLayout] = useBookerStore((state) => [state.layout, state.setLayout], shallow);
+  const [bookerState, setBookerState] = useBookerStore((state) => [state.state, state.setState], shallow);
+  const selectedDate = useBookerStore((state) => state.selectedDate);
+  const [selectedTimeslot, setSelectedTimeslot] = useBookerStore(
+    (state) => [state.selectedTimeslot, state.setSelectedTimeslot],
+    shallow
+  );
+
+  useBrandColors({
+    brandColor: event.data?.profile.brandColor,
+    darkBrandColor: event.data?.profile.darkBrandColor,
+  });
+
+  useInitializeBookerStore({
+    username,
+    eventSlug,
+    month,
+    eventId: event?.data?.id,
+    rescheduleUid,
+    rescheduleBooking,
+  });
+
+  useEffect(() => {
+    setLayout(isMobile ? "mobile" : "small_calendar");
+  }, [isMobile, setLayout]);
```

Comment:
Would this be better using `useLayoutEffect` ideally we just need the dom to have been loaded and not the whole component for this i believe?

Reply:
Using `useLayoutEffect` would actually do a first render that's visible to the user, and only then update the layout. So in theory that could result in first seeing layout A, and only after seeing the mobile layout? I'm wondering what the benefit of it would be?

Reply:
Youre right - think i was miss understanding what it does in my head :P 

Just re-read what it actually does - youre right this is not what we need